pkg/vdbgen/vdb.go

```diff
@@ -171,6 +180,30 @@ func (d *DBGenerator) setShardCount(ctx context.Context) error {
 	return nil
 }
 
+// setKsafety will fetch the ksafety from the database and set it inside v.vdb
+func (d *DBGenerator) setKSafety(ctx context.Context) error {
```

Comment:
We should update the e2e test for this case.  It currently is k-safety 0, but ends up having 3 nodes.  We likely recreated the vdb with k-safety of 1 and didn't even realize.  I suggest we had a few more steps to this test to scale down to 1 or 2 nodes and run vdb-gen again.